src/Xamarin.Android.Build.Utilities/Sdks/AndroidSdkBase.cs

```diff
@@ -81,6 +81,32 @@ public virtual void Initialize (string androidSdkPath = null, string androidNdkP
 				IsNdk64Bit = Directory.EnumerateDirectories (toolchainsDir, "arm-linux-androideabi-*")
 					.Any (dir => Directory.Exists (Path.Combine (dir, "prebuilt", NdkHostPlatform64Bit)));
 			}
+			// we need to look for extensions other than the default .exe|.bat
+			// google have a habbit of changing them.
+			foreach (var exe in Executables (Adb)) {
+				if (File.Exists (Path.Combine (AndroidPlatformToolsPath, exe)))
+					Adb = exe;
```

Comment:
This should presumably `break` out of the `foreach` once it's found, so that we don't look keep looking for non-existent files.

Additionally, this is all quite repetitive, so I'd suggest a helper:

    static string GetExecutablePath (string dir, string exe)
    {
        foreach (var e in Executables (exe))
            if (File.Exists (Path.Combine (dir, e))
                return e;
        return exe;
    }

    // ...
    Adb = GetExecutablePath (AndroidPlatformToolsPath, Adb);